### DISCUSSION TITLE
screen size

### DISCUSSION THREAD

**Discussion by chdelfosse** [epeios-q37/atlas-python#12] [General]
Is it possible to obtain current the screen's dimensions?
I know how to use css mediaqueries, but they are of no help
if one wants (for example)  to modify the "size" property of a select box.

I could do it in JavaScript (that and anything that mediaqueries do not cover),
but I do not see how to use JS together with the tookit.

**Comment by epeios-q37** [comment]
There are three functions to execute JS scripts: `dom.execute_(void|string|strings)(script)`. They are undocumented, because they are somewhat tricky to use regarding the returning of values, and they may fail silently when not properly used.

You best bet is to use `dom.execute_string(script)`, which returns a string, and to merge the various values you need into a string in a way that they can be easily extracted in *Python*. if `my_var` contains the string to return, or `my_func(…)` returns this string, put simply `my_var;` or `my_func(…);` as last statement of your JS script, **not** `return my_var;` or `return my_func(…);`

**Comment by chdelfosse** [comment]
I understand that (and why) the functions are undocumented.
I will try and use them starting with your recommendation.
I promise to try very hard before complaining or asking for more help.

Thanks for your help and for Atlastk as a tool.

Regards
ch

**Comment by chdelfosse** [comment]
I could not get dom.execute_string() to work.
However dom.execute_void() (with an example in Widgets/main.py) worked fine.
As to getting return values, setting some "data-...' attributes in an invisible `span` does the job satisfactorily.
From my point of view, this discussion may be closed.
Regards,
ch
